waku/protocol/v1/waku_mail.nim

```diff
@@ -76,8 +76,11 @@ proc requestMail*(node: EthereumNode, peerId: NodeId, request: MailRequest,
     return result
 
 proc p2pRequestHandler(peer: Peer, envelope: Envelope) =
-  # Mail server p2p request implementation
-  discard
+    let decoded = decode(envelope.data)
+    require decoded.isSome()
```

Comment:
require is only for testing, in order to immediately stop the tests on failure: https://nim-lang.org/docs/unittest.html#require.t%2Cuntyped . You have to properly check `if decoded.isSome`, to continue with its data.

(And in fact, require it is also broken for usage in tests and shouldn't be used at all, I see I forgot to remove its usage in nim-waku tests)